erts/emulator/nifs/common/prim_socket_nif.c

```diff
@@ -6888,6 +6898,34 @@ ERL_NIF_TERM esock_setopt_otp_iow(ErlNifEnv*       env,
 
 
 
+/* esock_setopt_otp_select_read - Handle the OTP (level) select_read option
+ */
+
+static
+ERL_NIF_TERM esock_setopt_otp_select_read(ErlNifEnv*       env,
+                                          ESockDescriptor* descP,
+                                          ERL_NIF_TERM     eVal)
+{
+    if (! IS_OPEN(descP->writeState)) {
```

Comment:
readState?

Reply:
Oops!  Cut and paste programming...  And I take both read and write mutexex, should be enough with the read mutex.

I also missed to implement esock_getopt_otp_select_read :-/